fetch/wire_types.go

```diff
@@ -48,22 +48,22 @@ func (r *MeshHashRequest) MarshalLogObject(encoder log.ObjectEncoder) error {
 }
 
 type MeshHashes struct {
-	Layers []types.LayerID
-	Hashes []types.Hash32
+	Layers []types.LayerID `scale:"max=1000"` // depends on syncer Config `MaxHashesInReq`, defaults to 100, 1000 is a safe upper bound
+	Hashes []types.Hash32  `scale:"max=1000"` // depends on syncer Config `MaxHashesInReq`, defaults to 100, 1000 is a safe upper bound
 }
 
 type MaliciousIDs struct {
-	NodeIDs []types.NodeID
+	NodeIDs []types.NodeID `scale:"max=100000"` // max. expected number of ATXs per epoch is 100_000
 }
 
 type EpochData struct {
-	AtxIDs []types.ATXID
+	AtxIDs []types.ATXID `scale:"max=1000000"`
 }
 
 // LayerData is the data response for a given layer ID.
 type LayerData struct {
-	Ballots []types.BallotID
-	Blocks  []types.BlockID
+	Ballots []types.BallotID `scale:"max=800"`
```

Comment:
i'll set this to 500 just to be consistent with number of proposals expected in a layer